### ISSUE TITLE
The ios-aarch64 platform is added on top of the original mac-aarch64

### ISSUE THREAD

**Issue by VisualEhrmanntraut** [Vector35/binaryninja-api#5684]
**Version and Platform (required):**
- Binary Ninja Version: 4.1.5637-dev
- OS: macOS
- OS Version: 15.0 Beta (24A5279h)
- CPU Architecture: M1

**Bug Description:**
I updated Binary Ninja and opened an iOS kernelcache bndb to resume work... waited and waited for the analysis... and... then noticed... huh???

<img width="783" alt="Screenshot 2024-07-03 at 23 07 16" src="https://github.com/Vector35/binaryninja-api/assets/30368284/f095a680-d11c-44d1-8243-02bc80208f78">

**Steps To Reproduce:**
Open an iOS bndb from a version before the iOS platform was added.

**Expected Behavior:**
It should migrate the old platform to the new one and remove the old one. 

**Screenshots/Video Recording:**
<img width="1552" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/30368284/91b4fadd-979d-4687-b0e4-b8df3472b01e">

**Binary:**
Can't, I don't think Apple would like that!

**Additional Information:**
The reanalysis has been on-going for 6+ hours now. This was not the case before. It might be analysing both platforms.

**Comment by emesare** [comment]
Can you print out the architecture and platform for both functions?
```
>>> current_function.arch
...
>>> current_function.platform
...
```
I suspect that the duplication is occurring due to some extra platform or architecture, possibly the new iOS platform.

**Comment by river-li** [comment]
Look at the left bottom corner, after 6 hours of reanalysis it's still Phase 1. I think this may relate to #5141

**Comment by VisualEhrmanntraut** [comment]
I left it overnight and it finished. Seems like it was trying to analyse some weird function:
```
[Function warn] Skipping analysis for function at 0xfffffff007dfd5b8 : Exceeds 'analysis.limits.maxFunctionAnalysisTime'
```

<img width="1552" alt="Screenshot 2024-07-04 at 08 35 05" src="https://github.com/Vector35/binaryninja-api/assets/30368284/21e2589b-2bfb-4ad2-9526-ab259b707481">

(which is duplicated again)

@emesare yes, it's indeed because of the new iOS platform.

```
>>> current_function.platform
<platform: ios-aarch64>
>>> current_function.platform
<platform: mac-aarch64>
```

**Comment by CouleeApps** [comment]
Proposed solution for this: in 4.1, update database version and upon loading any database created in a prior version, don't apply the new ios platform to auto-discovered functions.

**Comment by VisualEhrmanntraut** [comment]
Sounds more like a workaround than a solution, as that would make the old analysis databases not take advantage of the newly introduced platform.

**Comment by plafosse** [comment]
That is unfortunately correct. Apologies. Our existing database structure wasn't actually designed to cope with a new more specific platform being introduced and working with older databases. This issue was only discovered when you opened the issue.  Going forward we will have better options, but as of now the information just doesn't exist in the database to be able to make the correct decision.

**Comment by VisualEhrmanntraut** [comment]
Don't worry, I understand. I am also a software developer, I know how this stuff is. It should be easy to detect when the migration needs to happen though: if the ios-aarch64 platform is to be added, but the mac-aarch64 platform exists, migrate the important info from the old platform, put it in the newly added platform, remove the platform. Or something like that. Maybe that idea helps you guys get a bit unstuck. For now I would think it's best to just not add the new platform if the old one's here until a proper solution can be implemented.

**Comment by plafosse** [comment]
Fixed in 4.1.5729

**Comment by VisualEhrmanntraut** [comment]
@plafosse The issue got reintroduced somehow.

**Comment by emesare** [comment]
> @plafosse The issue got reintroduced somehow.

I can reproduce this by running "save-as" on the database then reopening. Is that what you did?

What does the following output for you?
```
>>> bv.file.database.globals[b"original_version"]
'6'
```

**Comment by VisualEhrmanntraut** [comment]
> I can reproduce this by running "save-as" on the database then reopening. Is that what you did?

@emesare Yes, I am scared to save over the original bndb until this is fixed properly.
